tests/graph/rewriting/test_nested.py

```diff
@@ -0,0 +1,94 @@
+from pytensor.compile.builders import OpFromGraph
+from pytensor.graph.fg import FunctionGraph
+from pytensor.graph.rewriting.basic import NodeRewriter
+from pytensor.graph.rewriting.nested import WalkingNestedGraphRewriter
+from pytensor.scan import scan
+from pytensor.tensor.elemwise import Elemwise
+from pytensor.tensor.math import sum
+from pytensor.tensor.type import matrix, scalar, vector
+
+
+class TestWalkingNestedGraphRewriter:
```

Comment:
We should do a more graph based test, and make sure the graph changed as we expected, and not just a counter variable that could be updated even without the changes working correctly